modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/data/provider/test/DDMRESTDataProviderTest.java

```diff
@@ -512,7 +507,136 @@ protected long saveDDMDataProviderInstance(
 		return dataProviderInstanceId;
 	}
 
-	protected void setPermissionCheckerUser(boolean guest) throws Exception {
+	private DDMFormValues _createDDMDataProviderDDMFormValues(
+		boolean cacheable, boolean filterable, String filterParameterName,
+		String inputParameterName, String outputParameterId,
+		String outputParameterPath, String outputParameterType,
+		String paginationEnd, String paginationStart, String url) {
+
+		DDMFormValues ddmFormValues = DDMFormValuesTestUtil.createDDMFormValues(
+			DDMFormFactory.create(_ddmDataProvider.getSettings()));
+
+		ddmFormValues.addDDMFormFieldValue(
+			DDMFormValuesTestUtil.createUnlocalizedDDMFormFieldValue(
+				"cacheable", String.valueOf(cacheable)));
+		ddmFormValues.addDDMFormFieldValue(
+			DDMFormValuesTestUtil.createUnlocalizedDDMFormFieldValue(
+				"filterable", String.valueOf(filterable)));
+		ddmFormValues.addDDMFormFieldValue(
+			DDMFormValuesTestUtil.createUnlocalizedDDMFormFieldValue(
+				"filterParameterName", filterParameterName));
+
+		if (Validator.isNotNull(inputParameterName)) {
+			DDMFormFieldValue inputParameters =
+				DDMFormValuesTestUtil.createDDMFormFieldValue(
+					"inputParameters", null);
+
+			ddmFormValues.addDDMFormFieldValue(inputParameters);
+
+			inputParameters.addNestedDDMFormFieldValue(
+				DDMFormValuesTestUtil.createUnlocalizedDDMFormFieldValue(
+					"inputParameterLabel", "input"));
+			inputParameters.addNestedDDMFormFieldValue(
+				DDMFormValuesTestUtil.createUnlocalizedDDMFormFieldValue(
+					"inputParameterName", inputParameterName));
+			inputParameters.addNestedDDMFormFieldValue(
+				DDMFormValuesTestUtil.createUnlocalizedDDMFormFieldValue(
+					"inputParameterRequired", Boolean.FALSE.toString()));
+			inputParameters.addNestedDDMFormFieldValue(
+				DDMFormValuesTestUtil.createUnlocalizedDDMFormFieldValue(
+					"inputParameterType", "[\"text\"]"));
```

Comment:
@rodrigopaulino, do you see value to create another test to cover the scenario in which the `inputParameterType` is of type **number**?

Reply:
I thought about it, but I didn't see a purpose for it, because, eventually, we're converting all "numbers" into "texts". I'm not entirely sure why there's an option for the type of inputs in the UI. I thought maybe this is relevant before we hit the `DDMRESTDataProvider` class, but certainly not in it. What do you think?

Reply:
I just checked the code and I agree with you. :smile: